src/main/java/com/gard/testbed/activitySystem/ActivityRunner.java

```diff
@@ -0,0 +1,51 @@
+package com.gard.testbed.activitySystem;
+
+/**
+ * Created by Chris on 16/03/2016..
+ */
+public class ActivityRunner implements IObserver {
+
+    // Blackboard for current player progress
+    IBlackboard player = null;
+    // Activity map for current activity
+    IActivity activity = null;
+    // Subscription to Subjects blackboard
+    private IObservable subject = null;
+
+    // TODO - list of active States
+    // ?? Maintain a list here or just rely on method (getActiveStates(player)) implemented in Activity ??
+
+    public ActivityRunner() {
+
+    }
+
+    // TODO - get start state and put in activeStates
+
+    // temp message method - replace with hook into message bus listening for update events
+    public void updateBlackboard(String key, int value) {
```

Comment:
With this method you are binding in the interface name how you implement thesolution why ? :)